examples/plugins/iowait/main.go

```diff
@@ -52,70 +53,188 @@ func main() {
 
 	plugin := &Plugin{HostID: *hostID}
 	http.HandleFunc("/report", plugin.Report)
+	http.HandleFunc("/control", plugin.Control)
 	if err := http.Serve(listener, nil); err != nil {
 		log.Printf("error: %v", err)
 	}
 }
 
 // Plugin groups the methods a plugin needs
 type Plugin struct {
-	HostID string
+	HostID     string
+	iowaitMode bool
 }
 
 // Report is called by scope when a new report is needed. It is part of the
 // "reporter" interface, which all plugins must implement.
 func (p *Plugin) Report(w http.ResponseWriter, r *http.Request) {
 	log.Println(r.URL.String())
-	now := time.Now()
-	nowISO := now.Format(time.RFC3339)
-	value, err := iowait()
+	metric, metricTemplate, err := p.metricsSnippets()
 	if err != nil {
 		log.Printf("error: %v", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	fmt.Fprintf(w, `{
+	topologyControl, nodeControl := p.controlsSnippets()
+	rpt := fmt.Sprintf(`{
 		 "Host": {
 			 "nodes": {
 				 %q: {
-					 "metrics": {
-						 "iowait": {
-							 "samples": [ {"date": %q, "value": %f} ],
-							 "min": 0,
-							 "max": 100
-						 }
-					 }
+					 "metrics": { %s },
+					 "controls": { %s }
 				 }
 			 },
-			 "metric_templates": {
-				 "iowait": {
-					 "id":       "iowait",
-					 "label":    "IO Wait",
-					 "format":   "percent",
-					 "priority": 0.1
-				 }
-			 }
+			 "metric_templates": { %s },
+			 "controls": { %s }
 		 },
 		 "Plugins": [
 			 {
 				 "id":          "iowait",
 				 "label":       "iowait",
 				 "description": "Adds a graph of CPU IO Wait to hosts",
-				 "interfaces":  ["reporter"],
+				 "interfaces":  ["reporter", "controller"],
 				 "api_version": "1"
 			 }
 		 ]
-	 }`, p.HostID+";<host>", nowISO, value)
+	 }`, p.getTopologyHost(), metric, nodeControl, metricTemplate, topologyControl)
+	fmt.Fprintf(w, "%s", rpt)
+}
+
+// Request is just a trimmed down xfer.Request
+type Request struct {
+	NodeID  string
+	Control string
+}
+
+// Control is called by scope when a control is activated. It is part
+// of the "controller" interface.
+func (p *Plugin) Control(w http.ResponseWriter, r *http.Request) {
+	log.Println(r.URL.String())
+	xreq := Request{}
+	err := json.NewDecoder(r.Body).Decode(&xreq)
 	if err != nil {
-		log.Printf("error: %v", err)
+		log.Printf("Bad request: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	thisNodeID := p.getTopologyHost()
+	if xreq.NodeID != thisNodeID {
+		log.Printf("Bad nodeID, expected %q, got %q", thisNodeID, xreq.NodeID)
+		w.WriteHeader(http.StatusBadRequest)
+		return
 	}
+	expectedControlID, _, _ := p.controlDetails()
+	if expectedControlID != xreq.Control {
+		log.Printf("Bad control, expected %q, got %q", expectedControlID, xreq.Control)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	p.iowaitMode = !p.iowaitMode
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprint(w, "{}")
+}
```

Comment:
As this is an example, maybe it's worth writing it in such as way that the plugin could handle multiple different controls? Maybe not worth the effort?